sound/soc/sof/intel/hda-stream.c

```diff
@@ -318,6 +318,18 @@ int hda_dsp_stream_hw_params(struct snd_sof_dev *sdev,
 	snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR, sd_offset,
 				SOF_HDA_CL_DMA_SD_INT_MASK |
 				SOF_HDA_SD_CTL_DMA_START, 0);
+
+	ret =
+	snd_sof_dsp_read_poll_timeout(sdev, HDA_DSP_HDA_BAR,
+				      sd_offset, run,
+				      ((run & SOF_HDA_SD_CTL_DMA_START)
```

Comment:
how is run set in this patch?

Reply:
the snd_sof_dsp_read_poll_timeout is a #define defined by @kv2019i :
set by line:
`(val) = snd_sof_dsp_read(sdev, bar, offset);   \`